table/filter.go

```diff
@@ -14,6 +14,15 @@ func (m Model) getFilteredRows(rows []Row) []Row {
 	filteredRows := make([]Row, 0)
 
 	for _, row := range rows {
+		if m.filterFunc != nil {
+			if m.filterFunc(row, filterInputValue) {
+				filteredRows = append(filteredRows, row)
+
+				// continue so we don't add the same row multiple times
+				continue
```

Comment:
This behavior means that the filter function AND the regular filtering is applied.  I think it would be clearer if the behavior is:

* If the filter func exists, then check with the filter func
* Otherwise, check isRowMatched

Someone using a filter func will want full control over their stuff, and sneakily also including literal text matches may not be what they want.